frontend/templates/product-list/index.tsx

```diff
@@ -70,6 +71,11 @@ export const getProductListServerSideProps = ({
          );
       }
 
+      const url = urlFromContext(context);
+      if (new URL(url).hostname !== PROD_HOSTNAME) {
+         context.res.setHeader('X-Robots-Tag', 'noindex, nofollow');
```

Comment:
Given that we're Just looking at a url and setting a header, should we do this in a middleware and not limit it to just the `product-list`?

Reply:
I hadn't considered that! I gave it a shot, but haven't used middlewares before, let me know what you think.